src/main/java/com/cloudbees/plugins/deployer/engines/Engine.java

```diff
@@ -181,7 +180,8 @@ public boolean perform() throws Throwable {
     public DeployedApplicationLocation process(FilePath applicationFile, T target) throws DeployException {
         try {
             FingerprintingWrapper actor = new FingerprintingWrapper(newDeployActor(target), listener);
-            return applicationFile.act(actor);
+            AbstractMap.SimpleEntry<String, DeployedApplicationLocation> pair = applicationFile.act(actor);
+            return this.addToFacects(pair);
```

Comment:
arguably more readable if `pair` is inlined (so the verbose type need not be mentioned)